File: app/src/main/java/me/devsaki/hentoid/activities/sources/ManhwaActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class ManhwaActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "manhwahentai.me";
    private static final String[] GALLERY_FILTER = {"//manhwahentai.me/[\\w\\-]+/[\\w\\-]{3,}/$"};
    private static final String[] DIRTY_ELEMENTS = {".c-ads"};
    private static final String[] JS_CONTENT_BLACKLIST = {"'iframe'","'adsdomain'","'closead'"};
    private static final String[] BLOCKED_CONTENT = {".cloudfront.net"};


    Site getStartSite() {
        return Site.MANHWA;
    }

    @Override
    protected CustomWebViewClient getWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        client.adBlocker.addToUrlBlacklist(BLOCKED_CONTENT);
        client.adBlocker.addJsUrlWhitelist(DOMAIN_FILTER);
        for (String s : JS_CONTENT_BLACKLIST) client.adBlocker.addJsContentBlacklist(s);
        client.addDirtyElements(DIRTY_ELEMENTS);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/ManhwaActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

class ManhwaActivity : BaseWebActivity() {
    companion object {
        const val GALLERY_PATTERN = "//manhwahentai.me/[%\\w\\-]+/[%\\w\\-]{3,}/$"

        private const val DOMAIN_FILTER = "manhwahentai.me"
        private val GALLERY_FILTER =
            arrayOf(GALLERY_PATTERN, GALLERY_PATTERN.replace("$", "") + "ch[%\\w]+-[0-9]+/$")
        private val REMOVABLE_ELEMENTS = arrayOf(".c-ads")
        private val BLOCKED_CONTENT = arrayOf(".cloudfront.net")
    }


    override fun getStartSite(): Site {
        return Site.MANHWA
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.adBlocker.addToUrlBlacklist(*BLOCKED_CONTENT)
        client.adBlocker.addToJsUrlWhitelist(DOMAIN_FILTER)
        client.addRemovableElements(*REMOVABLE_ELEMENTS)
        return client
    }
}